src/osuMania/systems/input.ts

```diff
@@ -30,35 +30,39 @@ export class InputSystem {
     document.addEventListener("keyup", this.handleKeyUp);
   }
 
+  private initKeybindsMap() {
+    const keybinds = this.game.settings.keybinds.keyModes[this.game.difficulty.keyCount - 1];
+    keybinds.forEach((key, index) => {
+      this.keybindsMap.set(key, index);
+    });
+  }
+
   public updateGamepadInputs() {
     const gamepad = navigator.getGamepads()[0];
 
-    if (!gamepad) {
-      return;
-    }
+    if (!gamepad) return;
 
-    for (let i = 0; i < gamepad.buttons.length; i++) {
-      const column = this.keybinds.indexOf(`🎮Btn${i}`);
-
-      if (column === -1) {
-        continue;
-      }
+    const updatedPressed = new Set<number>();
 
-      const button = gamepad.buttons[i];
+    gamepad.buttons.forEach((button, i) => {
+      const column = this.keybindsMap.get(`🎮Btn${i}`);
+      if (column === undefined) return;
 
       if (button.pressed && !this.gamepadState[i]?.pressed) {
         this.tappedColumns[column] = true;
         this.pressedColumns[column] = true;
-      }
-
-      if (!button.pressed && this.gamepadState[i]?.pressed) {
-        this.tappedColumns[column] = false;
-        this.pressedColumns[column] = false;
+        updatedPressed.add(column);
+      } else if (!button.pressed && this.gamepadState[i]?.pressed) {
         this.releasedColumns[column] = true;
+        updatedPressed.delete(column);
       }
-    }
+    });
 
     this.gamepadState = [...gamepad.buttons];
+
+    this.pressedColumns.forEach((_, i) => {
+      if (!updatedPressed.has(i)) this.pressedColumns[i] = false;
+    });
```

Comment:
Because of L63-L65, there's a bug where holds are immediately broken. The columns in `pressedColumns` are becoming false the frame after the column is pressed

Fix by adding L56 (this.pressedColumns[column] = false;) back into the else if block, and then `updatePressed` shouldn't be needed anymore